kernel/src/thread/mod.rs

```diff
@@ -46,9 +49,11 @@ impl Thread {
         priority: Priority,
         cpu_affinity: CpuSet,
     ) -> Self {
+        let catch_panics = !oops::panic_on_oops();
```

Comment:
This is a bit of surprise to me. Why is it more preferable to decide the panic-on-oops behavior upon the creation of a thread?